### PR TITLE
[Refactor] Change agg intermedaite state type to varbinary

### PR DESCRIPTION
**Pull request by LiShuMing**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/13880
- Change agg intermedaite state type to varbinary which later will be deduced as StreamMV IMTStateTable's table schema.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]

**Comment by LiShuMing** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java [hunk 1]
```diff
@@ -728,47 +712,12 @@ private void initAggregateBuiltins() {
...
-                        Type.VARCHAR,
-                        false, true, true));
-            } else if (t.isDecimalV2() || t.isDecimalV3()) {
-                addBuiltin(AggregateFunction.createBuiltin(FunctionSet.MULTI_DISTINCT_COUNT, Lists.newArrayList(t),
-                        Type.BIGINT,
-                        Type.VARCHAR,
-                        false, true, true));
-            }
+            // MULTI_DISTINCT_COUNTM
+            addBuiltin(AggregateFunction.createBuiltin(FunctionSet.MULTI_DISTINCT_COUNT, Lists.newArrayList(t),
```
Comment (ZiheLiu) [inline fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:716]: Maybe some complex types don't support `MULTI_DISTINCT_COUNT`, such as JSON/Map/Struct.

File: fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java [hunk 2]
```diff
@@ -1101,6 +896,159 @@ private void initAggregateBuiltins() {
...
+        addBuiltin(AggregateFunction.createBuiltin(FunctionSet.PERCENTILE_CONT,
+                Lists.newArrayList(Type.DATETIME, Type.DOUBLE), Type.DATETIME, Type.VARBINARY,
+                false, false, false));
+        addBuiltin(AggregateFunction.createBuiltin(FunctionSet.PERCENTILE_CONT,
+                Lists.newArrayList(Type.DOUBLE, Type.DOUBLE), Type.DOUBLE, Type.VARBINARY,
+                false, false, false));
+    }
+
+    private void registerBuiltinArrayAggFunction() {
+        addBuiltin(AggregateFunction.createBuiltin(FunctionSet.ARRAY_AGG,
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:1005]: why not write a for loop here ?

File: fe/fe-core/src/main/java/com/starrocks/catalog/Type.java [hunk 3]
```diff
@@ -133,23 +133,28 @@ public abstract class Type implements Cloneable {
     public static final Type ARRAY_VARCHAR = new ArrayType(Type.VARCHAR);
     public static final Type ARRAY_JSON = new ArrayType(Type.JSON);
 
-    private static final ImmutableList<ScalarType> INTEGER_TYPES =
-            ImmutableList.of(TINYINT, SMALLINT, INT, BIGINT, LARGEINT);
-    private static final ImmutableList<ScalarType> FLOAT_POINT_TYPES =
-            ImmutableList.of(FLOAT, DOUBLE, DECIMALV2, DECIMAL32, DECIMAL64, DECIMAL128);
+    public static final ImmutableList<ScalarType> INTEGER_TYPES =
+            ImmutableList.of(BOOLEAN, TINYINT, SMALLINT, INT, BIGINT, LARGEINT);
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Type.java:137]: why change `BOOLEAN` ?

File: fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java (hunk 1 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:716]: I'm afraid not. You need to implement it at BE at first, and each type need to define a `hash` function.

Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:716]: Therefore, we should judge whether the type `t` supports `MULTI_DISTINCT_COUNT` here.

File: fe/fe-core/src/main/java/com/starrocks/catalog/Type.java (hunk 3 above)
Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Type.java:137]: It's just like the behavior in `BE`, `BE` actually is UINT8Column?

File: fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java (hunk 1 above)
Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:716]: Yep, my mistake. I doubt why MULTI_COUNT_DISTINCT is different from `COUNT_DISTINCT`. I will revert to original policy.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 149 / 149 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|138|138|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|11|11|100.00%|[]|

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**